### ISSUE TITLE
Full text search for attachments

### ISSUE THREAD

**Issue by blaluc** [Mailu/Mailu#2184]
Hello,
is the full text search feature available for the message attachments, at least for the most common mediatype like pdf, docx, txt, etc.?
I've read something about xapian and solr but I didn't catch if the full-text is limited only to text fields (from, to, subject, body) or if there's a way to search also in the attachments.

**Comment by Diman0** [comment]
Looking at the docs (https://doc.dovecot.org/settings/plugin/fts-plugin/?highlight=fts#plugin_setting-fts-fts_decoder)
```fts_decoder = decode2text``` must be set in dovecot.conf to enable fts for attachments.

This is not enabled by default https://github.com/Mailu/Mailu/blob/master/core/dovecot/conf/dovecot.conf.

You could add an override for the dovecot configuration to enable this setting. 
See https://mailu.io/1.9/faq.html#how-can-i-override-settings for more information

**Comment by audioscavenger** [comment]
i could create a feature request but how does that work? outlook natively can do that so i would assume it's a client feature not related to devocot at all. is that feature needed for web clients only? and which web clients?

i enabled it along with the service needed as defined here https://doc.dovecot.org/settings/plugin/fts-plugin/#plugin_setting-fts-fts_decoder
- tested rainloop: does not work
- outlook: works with or without

can you guys provide some explanations so i can help, please?

**Comment by audioscavenger** [comment]
After careful research inside imap container, i found out that `fts_decoder = decode2text` + /usr/libexec/dovecot/decode2text.sh works indeed!

I sent myself TXT and DOCX attachements with keyword1, keyword2 and keyword3 to looks for something.
``` grep keyword /mail/me@domain.com/xapian-indexes/*
/mail/eric@domain.com/xapian-indexes/db_2de99a0bb5a1ec617d68000037366457/postlist.glass: XBDYkeyword1
/mail/eric@domain.com/xapian-indexes/db_2de99a0bb5a1ec617d68000037366457/postlist.glass: XBDYkeyword2
```

Outlook correctly returns emails with keyword1 in attachements, but neither roundcube or rainloop.
Are those webmails actually sending a SEARCH IMAP request?

**Comment by audioscavenger** [comment]
@willofr @Nebukadneza @tremlin hello guys, you implemented this with devocot, would you mind give me a hand when you have a minute? i appreciate, thanks a bunch!

**Comment by nextgens** [comment]
@audioscavenger I have implemented the basics on #2972 

Can you give it a test please?

Be careful, this is with master (and not stable): do backups or better yet try on a non-production environment
```
export MAILU_VERSION='pr-2972'
docker compose up -d
```

This uses catdoc and not lowriter

**Comment by audioscavenger** [comment]
> @audioscavenger I have implemented the basics on #2972
> 
> Can you give it a test please?
> 
> Be careful, this is with master (and not stable): do backups or better yet try on a non-production environment
> 
> ```
> export MAILU_VERSION='pr-2972'
> docker compose up -d
> ```
> 
> This uses catdoc and not lowriter

thank you so much! But sorry, i don't have a non-prod environment. I will simply wait for the next release. also I totally forgot about this. Also railoop is not used anymore, it's snappy now. not using the online search feature was the solution. Hopefully you can fix it one day :)
